Doc/library/array.rst

```diff
@@ -60,6 +60,13 @@ Notes:
    .. deprecated-removed:: 3.3 3.16
       Please migrate to ``'w'`` typecode.
 
+(2)
+   Int data types (signed or unsigned) can be 16 or 32 bits depending on the platform. The
+   same way that long data types can be 32 or 64 bits depending on the platform. On most
+   machines that run GNU C Library, an int is a 32-bit quantity. On most machines, long
+   int is also 32-bit, the same size as int. And lastly, on most machines, long long int
+   are 64-bit quantities. View more at: https://www.gnu.org/software/libc/manual/html_node/Range-of-Type.html
```

Comment:
I like the idea of using the otherwise nearly bare column to give better idea, but the proposed note is way too long.  I considered a shorter note based on Ronald's comment, but I want to propose something stripped to what I consider using the bare essentials that will cover 90+% of users, while saying more than I currently remembered.
```suggestion
   As of 2024, 'h', 'i', 'l', and 'q' are usually 2, 4, 4 on Windows else 8, and 8 bytes.
```
Note that the note will be followed by how to confirm the exact values for a particular machine with `.itemsize`.  If we can agree, I can finish this.  It is not worth hours of debate.

I also considered offering a code example,
```
from array import *
for t in 'h', 'i', 'l', 'q':
    print(f"{t} = {array(t).itemsize}")
```
but I think this is also more than is needed.

Reply:
in that case, ill commit the changes